### PR TITLE
Added hw-4

### PR DESCRIPTION
**Pull request by nkhamm-spb**

### CODE REVIEW

File: notebooks/hw-4.ipynb [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "cea29183",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "categorial_cols = ['model', 'transmission', 'fuelType']\n",
+    "data_x = data[categorial_cols]\n",
+    "data_y = data['price']\n",
+    "data_x = data_x.apply(LabelEncoder().fit_transform)"
```
Comment (svkov) [inline notebooks/hw-4.ipynb:151]: качество получилось плохое именно из-за label encoding, правильнее было бы взять one-hot

File: notebooks/hw-4.ipynb [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+   "outputs": [],
+   "source": [
+    "from sklearn.model_selection import GridSearchCV\n",
+    "\n",
+    "params = {\n",
+    "    \"criterion\": ('friedman_mse', 'squared_error', 'poisson', 'absolute_error'),\n",
+    "    \"max_depth\": list(range(3, 10)),\n",
+    "}\n",
+    "search = GridSearchCV(\n",
+    "    DecisionTreeRegressor(random_state=42),\n",
```
Comment (svkov) [inline notebooks/hw-4.ipynb:193]: А еще качество низкое, потому что с этими данными дерево плохо справляется
random forest с правильными фичами дает 0.98 по r2

**Review by svkov — APPROVED** [review]
зачтено
